### PR TITLE
Add circular redirect detection

### PR DESCRIPTION
**Pull request by pharman**
@micromax14

### CODE REVIEW

File: Validator/MapValidator.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        // Same from & to URLs within a map
+        if ($map->getUrlFrom() == $map->getUrlTo()) {
+            return false;
+        }
+
+        // Circular redirects
+        $self = $this;
+        $accumulator = [];
+        $recurse = function($map, $startPointMap) use ($self, &$recurse, &$accumulator) {
```
Comment (pharman) [inline Validator/MapValidator.php]: Type hint

File: Validator/MapValidator.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            $matchingToUrls = $this->mapRepository->findForUrlOrPath($map->getUrlTo(), $map->getUrlTo(), $accumulator);
```
Comment (pharman) [inline Validator/MapValidator.php]: What about cross-domain redirects?

File: Tests/Validator/MapValidatorTest.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+     * @var MockObject
+     */
+    private $mapRepository;
+
+    /**
+     * Initialise
+     */
+    public function setUp()
+    {
+        //echo get_class($mockEm->getRepository('Astina\Bundle\RedirectManagerBundle\Entity\Map'));exit;
```
Comment (micromax14) [inline Tests/Validator/MapValidatorTest.php]: forgot commented line

File: Tests/Validator/MapValidatorTest.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+namespace Astina\Bundle\RedirectManagerBundle\Tests\Validator;
+
+use Astina\Bundle\RedirectManagerBundle\Validator\MapValidator;
+use Astina\Bundle\RedirectManagerBundle\Entity\Map;
+
+/**
+ * Class MapValidatorTest
+ *
+ * @package   Astina\Bundle\RedirectManagerBundle\Tests\Validator
+ * @author    Matej Velikonja <mvelikonja@astina.ch>
```
Comment (micromax14) [inline Tests/Validator/MapValidatorTest.php:12]: You are the creator of this masterpiece!

**Comment by micromax14** [comment]
LGTM

**Pull request merged**